exporter/awss3exporter/s3_writer.go

```diff
@@ -57,6 +58,15 @@ func getS3Key(time time.Time, keyPrefix string, partition string, filePrefix str
 	return s3Key
 }
 
+func getAwsEndpoint(config *Config) string {
+	endpoint := config.S3Uploader.AwsEndpoint
+	if endpoint == "" {
+		endpoint = os.Getenv("AWS_ENDPOINT_URL")
```

Comment:
this is unexpected. we don't use env vars in the code like this, we have ways to inject env vars in our configuration.

Reply:
This environment variable is for configuring the AWS SDK.  It appears to be used here as a fallback.  A functionally-equivalent approach that does not require the collector to understand AWS SDK environment variable usage would be to only include the `Endpoint` field in the `aws.Config` used to create an SDK session when it is explicitly included in this configuration.